src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.h

```diff
@@ -124,10 +125,18 @@ struct FLTreeOnode final : Onode, Value {
       [&oroot](NodeExtentMutable &payload_mut, Recorder *recorder) {
 	auto &mlayout = *reinterpret_cast<onode_layout_t*>(
           payload_mut.get_write());
-	mlayout.omap_root.update(oroot);
+	Recorder::delta_op_t op;
+	if (oroot.get_type() == omap_type_t::OMAP_LARGE_LEAF) {
+	  mlayout.omap_root_large_leaf.update(oroot);
+	  op = Recorder::delta_op_t::UPDATE_OMAP_ROOT_LARGE_LEAF;
+	} else {
+	  assert(oroot.get_type() == omap_type_t::OMAP_SMALL_LEAF);
+	  mlayout.omap_root_small_leaf.update(oroot);
+	  op = Recorder::delta_op_t::UPDATE_OMAP_ROOT_SMALL_LEAF;
+	}
```

Comment:
To be consistent, either to merge `update_xattr_root()` here or split out another method to update the new omap root.